### PR TITLE
[SERV-239] Migrate to GH Actions

### PR DESCRIPTION
**Pull request by DRickard**
Convert from Travis CI to GitHub Actions CI

### CODE REVIEW

**Review by ksclarke — DISMISSED** [review]

File: scripts/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  echo "-rm ${SFTP_FILESPEC}"
+  # sftp will exit if put fails
+  echo "put target/${SFTP_FILESPEC}"
+  # Show a little extra info for logging
+  echo "pwd"
+  echo "ls -l"
+) | sftp -b - ${SFTP_USER}@${SFTP_SERVER}
+
+# Webhook to Jenkins server to initiate deploy
+curl "https://${CI_JENKINS_USER_CREDS}@${JENKINS_HOST}/${CI_JENKINS_JOB_URI}/buildWithParameters?token=${CI_JENKINS_DEPLOY_TOKEN}&TOMCAT_HOST=${CI_JENKINS_TOMCAT_HOST}&WEBAPP_NAME=${SFTP_PROJECT}"
```
Comment (markmatney) [inline scripts/deploy.sh]: It looks like several steps in the deploy are failing, leading to [no data being transferred](https://github.com/UCLALibrary/MusicReservesService/runs/5590744695?check_suite_focus=true#step:5:30).

**Comment by ksclarke** [comment]
Looks like my review is still good on here since there weren't any code changes in the FTP fix. 👍🏼

**Comment by markmatney** [comment]
Is it OK to re-run the GHA workflow?

File: scripts/deploy.sh (hunk 1 above)
Comment (markmatney) [inline scripts/deploy.sh]: Just realizing this, but this script should not exit 0 when it fails, because then the green checkmark is a lie.

**Review by markmatney — CHANGES_REQUESTED** [review]

**Comment by DRickard** [comment]
> 

Yes--the workflow deploys to TEST server.

**Comment by markmatney** [comment]
~So [the file exists](https://github.com/UCLALibrary/MusicReservesService/runs/5841413740?check_suite_focus=true#step:5:28), but is curl succeeding?~ Nevermind; I think the output with a bunch of zeros was misleading me

File: scripts/deploy.sh (hunk 1 above)
Comment (DRickard) [reply scripts/deploy.sh]: Added some error checks; do we want more than those?

Comment (markmatney) [reply scripts/deploy.sh]: That looks good to me

**Review by markmatney — CHANGES_REQUESTED** [review]
Looks like the workflow doesn't like that the log4j dependency declaration is missing a group ID.

**Review by ksclarke — APPROVED** [review]
Could you have just included a `set -e` after the `#!/bin/bash` (and used bash instead of sh since it's on the GA images) instead of the multiple `status=$?`? That would catch those earlier steps too, I think.

https://askubuntu.com/a/1303974/292301

**Comment by DRickard** [comment]
> Looks like the workflow doesn't like that the log4j dependency declaration is missing a group ID.

Just did a manual GHA run and it completed without errors, so... ¯\_(ツ)_/¯

**Comment by markmatney** [comment]
The build fails with the [same error as the PR build](https://github.com/UCLALibrary/MusicReservesService/runs/6083801502?check_suite_focus=true#step:4:13) when I run it locally with Maven 3.8.4:
```
$ mvn -ntp package
[INFO] Scanning for projects...
[ERROR] [ERROR] Some problems were encountered while processing the POMs:
[ERROR] 'dependencies.dependency.groupId' for null:log4j-core:jar is missing. @ line 30, column 17
 @ 
[ERROR] The build could not read 1 project -> [Help 1]
[ERROR]   
[ERROR]   The project Application1:MusicReserves:1.0-SNAPSHOT (/UCLALibrary/MusicReservesService/pom.xml) has 1 error
[ERROR]     'dependencies.dependency.groupId' for null:log4j-core:jar is missing. @ line 30, column 17
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
```

Can you please add
```xml
<groupId>org.apache.logging.log4j</groupId>
```
to the `log4j-core` declaration in pom.xml?

**Comment by DRickard** [comment]
> The build fails with the [same error as the PR build](https://github.com/UCLALibrary/MusicReservesService/runs/6083801502?check_suite_focus=true#step:4:13) when I run it locally with Maven 3.8.4:
> 
> ```
> $ mvn -ntp package
> [INFO] Scanning for projects...
> [ERROR] [ERROR] Some problems were encountered while processing the POMs:
> [ERROR] 'dependencies.dependency.groupId' for null:log4j-core:jar is missing. @ line 30, column 17
>  @ 
> [ERROR] The build could not read 1 project -> [Help 1]
> [ERROR]   
> [ERROR]   The project Application1:MusicReserves:1.0-SNAPSHOT (/UCLALibrary/MusicReservesService/pom.xml) has 1 error
> [ERROR]     'dependencies.dependency.groupId' for null:log4j-core:jar is missing. @ line 30, column 17
> [ERROR] 
> [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
> [ERROR] Re-run Maven using the -X switch to enable full debug logging.
> [ERROR] 
> [ERROR] For more information about the errors and possible solutions, please read the following articles:
> [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
> ```
> 
> Can you please add
> 
> ```
> <groupId>org.apache.logging.log4j</groupId>
> ```
> 
> to the `log4j-core` declaration in pom.xml?

Look at the pom in GitHub--the `groupID` tag is there.

**Comment by DRickard** [comment]
> The build fails with the [same error as the PR build](https://github.com/UCLALibrary/MusicReservesService/runs/6083801502?check_suite_focus=true#step:4:13) when I run it locally with Maven 3.8.4:
> 
> ```
> $ mvn -ntp package
> [INFO] Scanning for projects...
> [ERROR] [ERROR] Some problems were encountered while processing the POMs:
> [ERROR] 'dependencies.dependency.groupId' for null:log4j-core:jar is missing. @ line 30, column 17
>  @ 
> [ERROR] The build could not read 1 project -> [Help 1]
> [ERROR]   
> [ERROR]   The project Application1:MusicReserves:1.0-SNAPSHOT (/UCLALibrary/MusicReservesService/pom.xml) has 1 error
> [ERROR]     'dependencies.dependency.groupId' for null:log4j-core:jar is missing. @ line 30, column 17
> [ERROR] 
> [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
> [ERROR] Re-run Maven using the -X switch to enable full debug logging.
> [ERROR] 
> [ERROR] For more information about the errors and possible solutions, please read the following articles:
> [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/ProjectBuildingException
> ```
> 
> Can you please add
> 
> ```
> <groupId>org.apache.logging.log4j</groupId>
> ```
> 
> to the `log4j-core` declaration in pom.xml?

Also too: are you looking at the latest build, https://github.com/UCLALibrary/MusicReservesService/actions/runs/2191546763

**Comment by markmatney** [comment]
No, but that's not the one that gets auto-triggered by a PR. I'm looking at the one that's causing the "All checks have failed" status, which I re-ran about an hour ago.

I am not sure why the one you manually triggered is showing different results, though. That is odd.

**Comment by markmatney** [comment]
Er, it looks like a rebase of this branch is actually what's needed

**Review by markmatney — APPROVED** [review]

**Pull request merged**